### PR TITLE
feat: Improve message signing behaviour

### LINKED ISSUE #31: Fix multiple message signs
We don't need to have the users sign again and again when cookie is persisting.

### PR DESCRIPTION
**Pull request by QEDK**
Now uses the `authenticated` endpoint response to check cookie data and make signs more scarce.
Fixes #31

### CODE REVIEW

**Comment by Dhaiwat10** [comment]
I'm not seeing any message signing requests so I think this works? Haha

File: packages/dapp/src/contexts/Web3Provider.tsx [hunk 1]
```diff
@@ -211,25 +211,36 @@ const Web3Provider = ({ children }: { children: any }) => {
...
+        {
+          signature,
+        },
+        {
+          withCredentials: true,
+        }
+      );
+      if (verifyResponse.status !== 200) {
+        throw new Error("Unauthorized"); // handle this gracefully with an UI error
+      }
```
Comment (Dhaiwat10) [inline packages/dapp/src/contexts/Web3Provider.tsx:243]: Are we catching this error anywhere tho?

Comment (QEDK) [reply packages/dapp/src/contexts/Web3Provider.tsx:243]: Nope, but the idea is that we should beautify all errors when thrown.

**Comment by QEDK** [comment]
> I'm not seeing any message signing requests so I think this works? Haha

To test this out:
* Connect wallet, sign the message
* Disconnect it using MetaMask (don't click logout!)
* Reconnect using same wallet (testcase A): No message sign prompt
* Switch to a different wallet and reconnect (testcase B): Message sign prompt shows up

**Review by damianmarti — APPROVED** [review]
Great!

**Pull request merged**